### PR TITLE
Ch 24 - Add edge detection section

### PR DESCRIPTION
**Pull request by jasmaa**

### CODE REVIEW

**Comment by redblobgames** [comment]
BTW the 'vis' library seems to have moved as of a week ago to https://visjs.org / https://github.com/visjs

File: 1-Introduction/index.html [hunk 1]
```diff
@@ -6,9 +6,9 @@
   <script src="http://code.jquery.com/jquery-1.12.4.js"></script>
   <script type="text/javascript" src="../main.js"></script>
 </head>
-<body>
+<body class="flex-content-container">
```
Comment (samuelgoto) [inline 1-Introduction/index.html]: To keep this PR self contained to chapter 24, lets remove this change.

File: 24-Perception/Edge-Detection/index.html [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		<div class="content-container" id="content">
+			<div style="display: flex; flex-direction: column; row-gap: 1em;">
+
+				<h1>Edge Detection</h1>
+
+				<p>
+					For many of us, it is easy to find both the objects in a picture and their boundaries. Humans are
+					very good at identifying what something is and where it is given a clear line of sight, but what if
+					we need to find the boundaries of many objects in an enormously large number photos? While we could
+					put a room full of toddlers with markers at work and tell them to outline the edges in a stack of
```
Comment (samuelgoto) [inline 24-Perception/Edge-Detection/index.html]: The reference to toddlers and morals is unnecessary. Lets rephrase this without them.

Comment (samuelgoto) [reply 24-Perception/Edge-Detection/index.html]: I'd have this paragraph as:

`
For many of us, it is easy to find both the objects in a picture and their boundaries. Humans are very good at identifying what something is and where it is given a clear line of sight, but what if we need to find the boundaries of many objects in an enormously large number of photos? In computer vision, this is a general problem known as edge detection.
`

File: 24-Perception/Edge-Detection/index.html [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+					that it is able to reduce a complex image down to a simpler one where only edges are marked.
+					Because of this, edge detection often acts as a preliminary step to other computer vision processes.
+				</p>
+
+				<div id="pipeline2d-direct-root"></div>
+
+				<h2>Image Representation</h2>
+
+				<p>
+					Before jumping into edge detection, we still need to do a little work with images. Computers
```
Comment (samuelgoto) [inline 24-Perception/Edge-Detection/index.html]: I'd rephrease hits as:

"Before jumping into edge detection, lets first go over how computers represent images."

File: header.html [hunk 4]
```diff
@@ -1,77 +1,77 @@
...
+          <li><a href="/19-Knowledge-In-Learning">19. Knowledge in Learning</a></li>
+          <li><a href="/20-Learning-Probabilistic-Models">20. Learning Probabilistic Models</a></li>
+          <li><a href="/21-Reinforcement-Learning">21. Reinforcement Learning</a></li>
         </ul>
       </li>
       <li class="dropdown">
         <a class="dropdown-toggle" data-toggle="dropdown" href="#">Part 6
         <span class="caret"></span></a>
         <ul class="dropdown-menu">
-          <li><a href="../22-Natural-Language-Processing">22. Natural Language Processing</a></li>
```
Comment (samuelgoto) [inline header.html]: Why is this needed? Having the relative URLs help us decouple this from deployment?

File: styles.css [hunk 5]
```diff
@@ -33,3 +33,45 @@ pre{
 body > div.row {
     margin-right: 0px;
 }
+
+.flex-content-container {
```
Comment (samuelgoto) [inline styles.css]: I'm concerned about these global changes ... have you tested the rest of the pages with this?

File: header.html (hunk 4 above)
Comment (jasmaa) [reply header.html]: I removed this when I split the section into two pages originally. Relative url assumes we only inject the header in pages one directory down which breaks links if we try to add more pages per section. Some operations in my section require the site to be on a server to work so we can't really avoid decoupling...

File: styles.css (hunk 5 above)
Comment (jasmaa) [reply styles.css]: CSS here is mainly so it can trickle down through multiple directories for my section and for potential future use in other pages. The classes are currently not used in other pages.

File: 1-Introduction/index.html [hunk 6]
```diff
@@ -7,7 +7,7 @@
   <script type="text/javascript" src="../main.js"></script>
 </head>
 <body>
-
+
```
Comment (samuelgoto) [inline 1-Introduction/index.html]: revert?

File: styles.css (hunk 5 above)
Comment (samuelgoto) [reply styles.css]: can't it be included only in your chapter directory then (e.g. overriding the main css)?

File: main.js [hunk 7]
```diff
@@ -15,18 +15,53 @@
     $(function() {
         var body = document.getElementsByTagName('body')[0];
 
+		// Inject breadcrumbs
```
Comment (samuelgoto) [inline main.js]: I'm concerned about global changes here ... can't this be moved to your page only?
